libraries/Microsoft.Bot.Builder.Prompts/NumberPrompt.cs

```diff
@@ -45,6 +45,8 @@ protected NumberPrompt(IModel model, PromptValidator<NumberResult<T>> validator
             if (context.Request.Type != ActivityTypes.Message)
                 throw new InvalidOperationException("No Message to Recognize");
```

Comment:
>throw new InvalidOperationException("No Message to Recognize"); [](start = 16, length = 63)

I went back and forth on this and decided to just treat this as empty text. I didn't want to FORCE the developer into a situation where they HAVE to filter to message before calling prompts.